frontend/src/model/beta/reviews/ReviewDisplay.tsx

```diff
@@ -1,60 +1,67 @@
-
-  const reviewKindText = useMemo(() => (review.kind === 'release' ? 'release' : 'access request'), [review.kind])
+interface ReviewResponseWithRole extends ReviewResponse {
+  role: string
+}
 
-  const { acceptedReviewResponses, changesRequestedReviewResponses } = useMemo(() => {
-    return review.responses.reduce<{
-      acceptedReviewResponses: ReactElement[]
-      changesRequestedReviewResponses: ReactElement[]
-    }>(
-      (reviewResponses, reviewResponse) => {
-        const isAccepted = reviewResponse.decision === ApprovalStates.Accepted
+export default function ReviewDisplay({ reviews }: ReviewDisplayProps) {
+  const { modelRoles, isModelRolesLoading, isModelRolesError } = useGetModelRoles(
+    reviews[0] ? reviews[0].model.id : undefined,
+  )
 
-        const reviewResponseElement = (
-          <Stack direction='row' spacing={1} key={`${reviewResponse.user}-${reviewResponse.decision}`}>
-            {isAccepted ? (
-              <Done color='success' fontSize='small' />
-            ) : (
-              <HourglassEmpty color='warning' fontSize='small' />
-            )}
-            <Typography variant='caption'>
-              <Box component='span' fontWeight='bold'>
-                <UserDisplay dn={reviewResponse.user} />
-              </Box>
-              {` has ${isAccepted ? 'approved' : 'requested changes for'} this ${reviewKindText} (${getRoleDisplay(
-                review.role,
-                modelRoles,
-              )})`}
-            </Typography>
-          </Stack>
-        )
+  const [acceptedReviewResponses, setAcceptedReviewResponses] = useState<ReviewResponseWithRole[]>([])
+  const [changesRequestedReviewResponses, setChangesRequestedReviewResponses] = useState<ReviewResponseWithRole[]>([])
 
-        if (isAccepted) reviewResponses.acceptedReviewResponses.push(reviewResponseElement)
-        else reviewResponses.changesRequestedReviewResponses.push(reviewResponseElement)
+  useEffect(() => {
+    const updatedApprovals: ReviewResponseWithRole[] = []
+    const updatedRequests: ReviewResponseWithRole[] = []
+    if (reviews) {
+      reviews.forEach((review) =>
+        review.responses.forEach((reviewResponse) => {
+          reviewResponse.decision === 'approve'
+            ? updatedApprovals.push({ ...reviewResponse, role: review.role })
+            : updatedRequests.push({ ...reviewResponse, role: review.role })
+        }),
+      )
+      setAcceptedReviewResponses(updatedApprovals)
+      setChangesRequestedReviewResponses(updatedRequests)
+    }
+  }, [reviews])
 
-        return reviewResponses
-      },
-      { acceptedReviewResponses: [], changesRequestedReviewResponses: [] },
-    )
-  }, [review, modelRoles, reviewKindText])
+  const approvalsDisplay = () => {
+    const staticRoles = ['owner', 'contributor', 'consumer']
+    const dynamicRoles = modelRoles.filter((role) => !staticRoles.includes(role.id)).map((role) => role)
+    const roleApprovals: string[] = []
```

Comment:
This would be `useMemo`able if you make this function into a real component. The logic for 
`staticRoles` and `dynamicApprovals` could be included in the same `useMemo`